ports/rp2/mpnetworkport.c

```diff
@@ -147,7 +170,16 @@ void mod_network_lwip_init(void) {
         mp_network_soft_timer_callback
         );
 
-    soft_timer_reinsert(&mp_network_soft_timer, LWIP_TICK_RATE_MS);
+    if (netif_callback.callback_fn == NULL) {
+        netif_add_ext_callback(&netif_callback, mp_network_netif_status_cb);
+    }
+}
+
+static void mp_network_netif_status_cb(struct netif *netif, netif_nsc_reason_t reason, const netif_ext_callback_args_t *args) {
+    // Start the network soft timer any time an interface comes up
+    if (reason == LWIP_NSC_LINK_CHANGED && args->link_changed.state) {
+        soft_timer_reinsert(&mp_network_soft_timer, LWIP_TICK_RATE_MS);
```

Comment:
Maybe this needs a check `... && mp_network_soft_timer.mode == SOFT_TIMER_MODE_ONE_SHOT` so that it's only reinserted if it was disabled.

That would prevent pathological cases like an interface going up/down continuously and constantly pushing the timer out further and further.

Reply:
Check added. Also needed to switch the `mode` here to avoid the pathological case (otherwise too-fast interface flapping can still starve the timeout).